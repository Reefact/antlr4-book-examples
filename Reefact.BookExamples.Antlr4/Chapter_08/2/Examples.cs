#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_08._2 {

    [UseReporter(typeof(CustomReporter))]
    public class Examples {

        [Fact]
        public void translate() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.json", 8, 2);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string result = grun.Translate();
            // Verify
            Approvals.Verify(result);
        }

    }

}