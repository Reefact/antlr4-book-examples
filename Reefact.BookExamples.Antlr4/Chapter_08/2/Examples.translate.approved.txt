
<description>An imaginary server config file</description>
<logs>
<level>verbose</level>
<dir>/var/log</dir>
</logs>
<host>antlr.org</host>
<admin>
<element>
<element>parrt</element>
<element>pirrt</element>
</element>
<element>tombu</element>
</admin>
<aliases></aliases>
